projects/ui-policy-studio-angular/src/lib/environment-flow-studio/gio-environment-flow-studio.component.ts

```diff
@@ -13,65 +13,119 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { Component, EventEmitter, Input, OnChanges, Output, SimpleChanges } from '@angular/core';
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, OnChanges, Output, SimpleChanges } from '@angular/core';
 import { EMPTY } from 'rxjs';
-import { GioFormJsonSchemaModule } from '@gravitee/ui-particles-angular';
+import { GioFormJsonSchemaModule, GioLoaderModule } from '@gravitee/ui-particles-angular';
 
 import { GioPolicyStudioDetailsPhaseComponent } from '../components/flow-details-phase/gio-ps-flow-details-phase.component';
-import { ApiType, ConnectorInfo, Policy, PolicyDocumentationFetcher, PolicySchemaFetcher, Step } from '../models';
+import { ApiType, ConnectorInfo, ExecutionPhase, Policy, PolicyDocumentationFetcher, PolicySchemaFetcher, Step } from '../models';
 import { GioPolicyStudioService } from '../policy-studio/gio-policy-studio.service';
 
 export type EnvironmentFlowOutput = Step[];
 
+type PhaseVM = {
+  name: string;
+  description: string;
+};
+
 @Component({
   selector: 'gio-environment-flow-studio',
   standalone: true,
-  imports: [GioPolicyStudioDetailsPhaseComponent, GioFormJsonSchemaModule],
+  imports: [GioPolicyStudioDetailsPhaseComponent, GioFormJsonSchemaModule, GioLoaderModule],
   templateUrl: './gio-environment-flow-studio.component.html',
   styleUrl: './gio-environment-flow-studio.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class GioEnvironmentFlowStudioComponent implements OnChanges {
+  /**
+   * List of policies usable in the studio
+   */
   @Input()
   public policies: Policy[] = [];
 
+  /**
+   * Type of the API targeted by the environment flow
+   */
+  @Input({ required: true })
+  public apiType!: ApiType;
+
+  /**
+   * Execution phase targeted by the environment flow
+   */
+  @Input({ required: true })
+  public executionPhase!: ExecutionPhase;
+
   /**
    * Called when Environment-flow Studio needs to fetch the policy schema
    * @returns Observable of the policy schema
    */
-  @Input()
+  @Input({ required: true })
   public policySchemaFetcher: PolicySchemaFetcher = () => EMPTY;
 
   /**
    * Called when Environment-flow Studio needs to fetch the policy documentation
    * @returns Observable of the policy documentation
    */
-  @Input()
+  @Input({ required: true })
   public policyDocumentationFetcher: PolicyDocumentationFetcher = () => EMPTY;
 
+  /**
+   * Output event when the environment flow changes
+   */
   @Output()
   public environmentFlowChange: EventEmitter<EnvironmentFlowOutput> = new EventEmitter<EnvironmentFlowOutput>();
 
+  protected isLoading = true;
   protected readOnly = false;
   protected startConnector: ConnectorInfo[] = [
     {
       name: 'Incoming request',
```

Comment:
related to my previous comment, I think we should update name here depending on the selected phase